pkg/cmd/controller/serviceaccount.go

```diff
@@ -1,86 +1,56 @@
 package controller
 
 import (
-	"k8s.io/klog/v2"
-
-	kapiv1 "k8s.io/api/core/v1"
-	sacontroller "k8s.io/kubernetes/pkg/controller/serviceaccount"
+	"fmt"
 
-	serviceaccountcontrollers "github.com/openshift/openshift-controller-manager/pkg/serviceaccounts/controllers"
+	openshiftcontrolplanev1 "github.com/openshift/api/openshiftcontrolplane/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+	"k8s.io/kubernetes/pkg/controller/serviceaccount"
 )
 
 func RunServiceAccountController(ctx *ControllerContext) (bool, error) {
-	if len(ctx.OpenshiftControllerConfig.ServiceAccount.ManagedNames) == 0 {
-		klog.Infof("Skipped starting Service Account Manager, no managed names specified")
+	managedNames := ctx.OpenshiftControllerConfig.ServiceAccount.ManagedNames
+	if len(managedNames) == 0 {
+		klog.Info(openshiftcontrolplanev1.OpenShiftServiceAccountController + ": no managed names specified")
 		return false, nil
 	}
+	return runServiceAccountsController(ctx, managedNames...)
+}
+
+func RunBuilderServiceAccountController(ctx *ControllerContext) (bool, error) {
+	return runServiceAccountsController(ctx, "builder")
+}
 
-	options := sacontroller.DefaultServiceAccountsControllerOptions()
-	options.ServiceAccounts = []kapiv1.ServiceAccount{}
+func RunDeployerServiceAccountController(ctx *ControllerContext) (bool, error) {
+	return runServiceAccountsController(ctx, "deployer")
+}
 
-	for _, saName := range ctx.OpenshiftControllerConfig.ServiceAccount.ManagedNames {
-		// the upstream controller does this one, so we don't have to
-		if saName == "default" {
+func runServiceAccountsController(cctx *ControllerContext, managedNames ...string) (bool, error) {
+	options := serviceaccount.DefaultServiceAccountsControllerOptions()
+	options.ServiceAccounts = nil
+	for _, name := range managedNames {
+		if name == "default" {
+			// kube-controller-manager already does this one
 			continue
 		}
-		sa := kapiv1.ServiceAccount{}
-		sa.Name = saName
-
-		options.ServiceAccounts = append(options.ServiceAccounts, sa)
+		options.ServiceAccounts = append(options.ServiceAccounts,
+			corev1.ServiceAccount{ObjectMeta: metav1.ObjectMeta{Name: name}},
+		)
 	}
-
-	controller, err := sacontroller.NewServiceAccountsController(
-		ctx.KubernetesInformers.Core().V1().ServiceAccounts(),
-		ctx.KubernetesInformers.Core().V1().Namespaces(),
-		ctx.ClientBuilder.ClientOrDie(infraServiceAccountControllerServiceAccountName),
-		options)
-	if err != nil {
-		return true, nil
+	if len(options.ServiceAccounts) == 0 {
```

Comment:
Nit: fail fast, iow. this should be the first check in `runServiceAccountsController` method.

Reply:
It is down here to also catch the case where `len(managedNames) >0` and they are all `"default"`.